### DISCUSSION TITLE
Proper installer for Windows

### DISCUSSION THREAD

**Discussion by sryze** [usebruno/bruno#2281] [Ideas]
Please make it install to `C:\Program Files` instead of %APPDATA% like ~normal software~ the majority of (classic) software for Windows.

Currently when a user runs the Bruno installer, it silently installs itself to the user's AppData directory without asking/confirming, which may be unexpected (I personally would expect to be offered some options or an additional dialog with an "Install" button before proceeding). In addition to that, many users expect to find the files installed by various software in the common location (e.g. if they need to access them for some reason or perform some administration tasks).

**Comment by qurm** [comment]
Similarly, I have had issues with Windows install.  I installed using Chocolatey, as per the option on the download.  I had to install as Admin (common on Windows), however the bruno shortcut theni ends up in the admin users Desktop and  %APPDATA% folder, with no explanation to the user.    Note that one of the reasons it is standard to install in `Program Files` is to ensure that the app is available for all users. 

At least this needs to be documented [here](https://www.usebruno.com/downloads) for the windows install to avoid a good deal of frustration.  i.e. the exe path is `C:\Users\myuser\AppData\Local\Programs\bruno\Bruno.exe`

(Yes,  part of the issue here is Chocolatey behaviour, but that is just the installer, and is reasonable to run as Admin)

**Reply by danielniccoli** [reply]
> I had to install as Admin (common on Windows)

There is a difference between changing user context (what you did) and gaining elevated permissions via UAC. One does change the user context, and the other just requests additional privileges.

You did the former. This is confusing, yes. But there is no way around that, if Bruno installed is meant to install the application into  `FOLDERID_UserProgramFiles`[^1].


[^1]: https://learn.microsoft.com/en-us/windows/win32/shell/knownfolderid [↩](https://github.com/usebruno/bruno/discussions/2281#user-content-fnref-1-1f1bff37843a02521619acb1c484d9e5)

**Reply by qurm** [reply]
I agree about UAC, and the install location is acceptable.  

However it is the silent install in an unexpected location, without any message, and without any mention on the installation page documentation that make this a difficult user experience.   A message showing the install path, or an Install for all Users option would both have helped to understand and avoid the issue.

**Comment by danielniccoli** [comment]
> Currently when a user runs the Bruno installer, it silently installs itself to the user's AppData directory without asking/confirming, which may be unexpected (I personally would expect to be offered some options or an additional dialog with an "Install" button before proceeding).

I agree that the installer should require a confirmation via the click of a button, before installing the application.

> In addition to that, many users expect to find the files installed by various software in the common location

`%LOCALAPPDATA%\Programs` is a common location. Microsoft gave it its own "Known Folder ID" `FOLDERID_UserProgramFiles`[^1].

![grafik](https://github.com/usebruno/bruno/assets/2971735/8c2a18bf-37d8-4d20-aae6-5b28236ba9c0)

I think it's a great location to install smaller software, that is typically not shared between users of a single device, and prefer this to be the default installation folder. However, I don't see any downsides of having an option to install Bruno for all users.




[^1]: https://learn.microsoft.com/en-us/windows/win32/shell/knownfolderid

**Comment by igorakkerman** [comment]
The installer also seems to start the application after installing or updating it. I use WinGet to update all installed applications automatically and don't want them to start after they have been updated.

**Comment by end3rbyte** [comment]
It's getting rare to have computer admin access in big companies.
In my case, installing in C:\Program Files is not an option.
Give the choice on the location, why not.

**Comment by bmirkic** [comment]
It would be excellent to know if there is a location switch that can be passed to the installer. In my case, with Chocolatey for Business, Bruno Installer wants to install it in some funny folder where Chocolatey for Business cannot write. The outcome is that Chocolatey says the software is installed, but the installation never happens, and there is no exe anywhere on the machine.

**Comment by LenhardErwan** [comment]
I face the same problem.

In my context a specific user with admin rights is used to do all installations via chocolatey. But as a result, I end up with bruno exe in this user's APPDATA directory, except that obviously from my user, I can't navigate down there.

It would be really nice to be able to use an argument to choose where the application is installed.

**Comment by Pragadesh-45** [comment]
**@bmirkic @LenhardErwan** Hey, can you try this to install the Bruno Chocolatey package in a custom path?

```powershell
choco install bruno -y --ia "/D=C:\CustomInstallPath"
```

**Reply by LenhardErwan** [reply]
So we're limited with Chocolatey and we're in background mode, which prevents the use of ‘install-arguments’.

**Comment by Pragadesh-45** [comment]
And for `.exe` installers, we already support custom installation paths. Can anyone in this thread confirm this?

**Reply by LenhardErwan** [reply]
With `.exe` installer, yes we can choose installation path. But I don't know if it's possible to launch the executable with particular parameters to choose directly from the terminal.

**Comment by stevericks** [comment]
msiexec.exe /i "bruno.msi" /qn ALLUSERS=1 /log c:\temp\install_bruno.log

**Comment by Anonymenet77** [comment]
Hello,

I am following up on this matter as we are facing a critical issue with the deployment of your software within our company. The current installation does not meet the expected standards:

The software does not install in the C:\Program Files (or Program Files (x86)) directory, which is a requirement for our infrastructure.
It also does not appear in the list of installed programs in the Control Panel, making it difficult to manage and maintain.

We use SCCM + PADT for deployment, and these discrepancies prevent us from achieving a clean and compliant integration. Given that we have invested in a significant number of licenses, this issue calls into question the viability of the solution for our environment.

**Urgent Question:**
Is there a plan to address these issues in an upcoming release to align the installation with industry standards (installation path and visibility in the list of installed programs)? If so, could you provide us with an estimated timeline?
Without a prompt resolution, we may unfortunately have to reconsider our decision and explore alternative solutions, which would be regrettable for both parties.

I am available to discuss the technical details or our team’s specific requirements. Please provide feedback as soon as possible so we can plan our next steps accordingly.

Thank you for your attention to this matter.

Best regards,
